backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/DiagramService.java

```diff
@@ -60,12 +61,23 @@ public DiagramService(IRepresentationRepository representationRepository, Object
 
     @Override
     public Diagram create(DiagramCreationParameters parameters) {
+        return this.create(parameters, Optional.empty());
+    }
+
+    @Override
+    public Diagram refresh(DiagramCreationParameters parameters, IDiagramRefreshManager refreshManager) {
+        return this.create(parameters, Optional.of(refreshManager));
+    }
+
+    private Diagram create(DiagramCreationParameters parameters, Optional<IDiagramRefreshManager> optionalRefreshManager) {
         VariableManager variableManager = new VariableManager();
         variableManager.put(GetOrCreateRandomIdProvider.PREVIOUS_REPRESENTATION_ID, parameters.getId());
         variableManager.put(DiagramDescription.LABEL, parameters.getLabel());
         variableManager.put(VariableManager.SELF, parameters.getObject());
         variableManager.put(IEditingContext.EDITING_CONTEXT, parameters.getEditingContext());
-
+        if (optionalRefreshManager.isPresent()) {
+            variableManager.put(IDiagramRefreshManager.DIAGRAM_REFRESH_MANAGER, optionalRefreshManager.get());
+        }
```

Comment:
Either the diagram manager is necessary or not. It may very well be necessary but in this case it should not be optional. This first commit is definitely missing an ADR to explain what you are trying to accomplish.